src/io/sb3/fromSb3.ts

```diff
@@ -441,9 +386,9 @@ export async function fromSb3JSON(json: sb3.ProjectJSON, options: { getAsset: ge
           async (spriteData: sb3.Sprite) =>
             new Sprite({
               name: spriteData.name,
-              costumes: await extractCostumes(spriteData, options.getAsset),
+              costumes: await extractCostumes(spriteData, options.getAssetData),
               costumeNumber: spriteData.currentCostume,
-              sounds: await extractSounds(spriteData, options.getAsset),
+              sounds: await extractSounds(spriteData, options.getAssetData),
               scripts: Object.entries(spriteData.blocks)
                 .filter(([id, block]) => block.topLevel)
                 .filter(([id, block]) => !block.shadow)
```

Comment:
same thing w.r.t "half the filtering, twice the performance"

Reply:
in fact, is there any way to de-duplicate this code with the above (`Object.entries(stageData.blocks)` and so forth)?

Reply:
Absolutely! I made a function to share all the common properties between the stage and sprites.